src/main/java/data/task/TaskList.java

```diff
@@ -0,0 +1,296 @@
+                } else {
+                    task = new EventTask(taskArray[2], taskArray[3], taskArray[4]);
+                }
+                if ((taskArray[1]).equals("1")) {
+                    task.setDone();
+                }
+                this.tasks.add(task);
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("An error occurred when trying to find the file.");
+            e.getMessage();
+            e.printStackTrace();
+        } catch (DukeException e) {
+            System.out.println(e);
+        }
+    }
+
+    /**
+     * Calls a specific method with the specified input based on the specified command.
+     *
+     * @param command The command to be run.
+     * @param input The input string for the command.
+     * @throws DukeException If an EventTask is instantiated with invalid dates.
+     */
+    public void executeCommand(Command command, String input) throws DukeException {
+        switch (command) {
+        case MARK:
+            this.setTaskComplete(input);
+            break;
+        case UNMARK:
+            this.setTaskIncomplete(input);
+            break;
+        case DELETE:
+            this.deleteTask(input);
+            break;
+        case TODO:
+        case DEADLINE:
+        case EVENT:
+            this.addTask(command, input);
+            break;
+        }
+    }
+
+    /**
+     * Adds a specified task to the ArrayList.
+     *
+     * @param task The task to be added to the ArrayList.
+     */
+    public void addTask(Task task, boolean isMuted) {
+        this.tasks.add(task);
+        if (!isMuted) {
+            System.out.println("Got it. I've added this task:");
+            System.out.println(task);
+            System.out.println("Now you have " + this.tasks.size() + " task(s) in the list.");
+        }
+    }
+
+    /**
+     * Adds a task to the ArrayList based on the specified command and input.
+     *
+     * @param command The command that determines which subclass of Task will be instantiated.
+     * @param input The input needed for the instantiation of the Task.
+     * @throws DukeException If the EventTask is instantiated with invalid dates.
+     */
+    public void addTask(Command command, String input) throws DukeException {
+        Task taskToAdd;
+        String[] inputArr;
+
+        switch (command) {
+        case TODO:
+            input = input.substring(5);
+            taskToAdd = new ToDoTask(input);
+            break;
+        case DEADLINE:
+            input = input.substring(9);
+            inputArr = input.split(" /by ");
+            taskToAdd = new DeadlineTask(inputArr[0], inputArr[1]);
+            break;
+        case EVENT:
+            input = input.substring(6);
+            inputArr = input.split(" /from ");
+            String description = inputArr[0];
+            inputArr = inputArr[1].split(" /to ");
```

Comment:
Maybe you can consider using different variables names instead of overwriting them?

Reply:
Will do, thanks!